src/controllers/router-store/index.tsx

```diff
@@ -200,6 +211,38 @@ const actions: AllRouterActions = {
 
     return { query, route, match };
   },
+
+  pushQueryParam: params => ({ getState }) => {
+    const { query: existingQueryParams, history, location } = getState();
+    const updatedQueryParams = { ...existingQueryParams, ...params };
+    // remove null/undefined keys
+    Object.keys(updatedQueryParams).forEach(
+      key =>
+        (updatedQueryParams[key] === undefined ||
+          updatedQueryParams[key] === null) &&
+        delete updatedQueryParams[key]
+    );
+    const updatedPath = updateQueryParams(
+      location,
+      updatedQueryParams as Query
+    );
+
+    history.push(getRelativePath(updatedPath) as any);
+  },
+
+  pushPathParam: params => ({ getState }) => {
+    const {
+      history,
+      location,
+      route: { path: rawPath },
+      match: { params: existingPathParams },
+    } = getState();
+    const updatedPathParams = { ...existingPathParams, ...params };
+    const updatedPath = generatePathUsingPathParams(rawPath, updatedPathParams);
+    const updatedLocation = { ...location, pathname: updatedPath };
+
+    history.push(getRelativePath(updatedLocation) as any);
```

Comment:
Can you explain why we need to call `getRelativePath` ? `updatedLocation` is an object so it should do nothing with it 🤔

Reply:
Our `history.push()` takes only string as param. `getRelativePath` util deconstructs the `Location` object and safely returns the url constructed in a string format.

Reply:
We have a new util now `getURLFromLocation` 😄